worldedit-core/src/main/java/com/sk89q/worldedit/extension/factory/parser/DefaultBlockParser.java

```diff
@@ -163,58 +163,56 @@ private String woolMapper(String string) {
     private static Map<Property<?>, Object> parseProperties(BlockType type, String[] stateProperties, ParserContext context) throws InputParseException {
         Map<Property<?>, Object> blockStates = new HashMap<>();
```

Comment:
Maybe change this to use `ImmutableMap`?

Reply:
I've made it wrap on return - can't use a builder as this does a `containsKey` check which builders don't support